app/javascript/widget/composables/useDarkMode.js

```diff
@@ -0,0 +1,31 @@
+import { computed } from 'vue';
+import { useMapGetter } from 'dashboard/composables/store';
+
+/**
+ * Composable for handling dark mode.
+ * @returns {Object} An object containing computed properties and methods for dark mode.
+ */
+
+export function useDarkMode() {
+  const darkMode = useMapGetter('appConfig/darkMode');
+  const prefersDarkMode = computed(() => {
+    const isOSOnDarkMode =
+      darkMode.value === 'auto' &&
+      window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Lets memoize the `window.matchMedia `call to avoid repeated DOM access.
```js
const memoizedMatchMedia = computed(() => window.matchMedia?.('(prefers-color-scheme: dark)').matches ?? false);
const prefersDarkMode = computed(() => 
  (darkMode.value === 'auto' && memoizedMatchMedia.value) || darkMode.value === 'dark'
);
```